fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ExtractSingleTableExpressionFromDisjunction.java

```diff
@@ -50,24 +52,44 @@
  * 3. In old optimizer, there is `InferFilterRule` generates redundancy expressions. Its Nereid counterpart also need
  * `RemoveRedundantExpression`.
  * <p>
- * TODO: This rule just match filter, but it could be applied to inner/cross join condition.
  */
-public class ExtractSingleTableExpressionFromDisjunction extends OneRewriteRuleFactory {
+public class ExtractSingleTableExpressionFromDisjunction implements RewriteRuleFactory {
+    private static final ImmutableSet<JoinType> ALLOW_JOIN_TYPE = ImmutableSet.of(JoinType.INNER_JOIN,
+            JoinType.LEFT_OUTER_JOIN, JoinType.RIGHT_OUTER_JOIN, JoinType.LEFT_SEMI_JOIN, JoinType.RIGHT_SEMI_JOIN,
```

Comment:
why not support full outer join?

Reply:
full outer join on condition predicate cannot be pushdown